### PR TITLE
r_d13: Wordle game

### PR DESCRIPTION
**Pull request by ShekhaninKyryl**

### CODE REVIEW

File: r_d13/utils.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            tempFile << word + ':' + currentDate + ':' + std::to_string(tries) << std::endl;
+            lineReplaced = true;
+        }
+        else {
+            tempFile << line << std::endl;
+        }
+    }
+
+    if (lineReplaced) {
+        //TODO: fix removing file
```
Comment (ShekhaninKyryl) [inline r_d13/utils.cpp:73]: can't remove file

File: r_d13/utils.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+enum class TypeOfWord
+{
+	WordOfTheDay,
+	RandomWord,
+	Exit,
+};
+
+std::string readFromFile(const std::string filePath, int lineNumber, std::string currentDate);
+bool writeToFile(const std::string filePath, std::string word, std::string currentDate, int tries);
```
Comment (bohdanpc) [inline r_d13/utils.h:11]: const ref

File: r_d13/utils.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+void getWord(TypeOfWord type, char*& wordSource, const std::string filePath);
+bool compareWords(char* word, char* userInput, char* result);
+
+bool isSymbolExist(char* word, char symbol, unsigned short position, char*& result, bool* ignorePosition);
+bool isSymbolOnPlace(char* word, char symbol, unsigned short position, char*& result);
```
Comment (bohdanpc) [inline r_d13/utils.h:17]: `char*&` ? Hmmm, I'm quite skeptic you really need this construct

File: r_d13/utils.cpp [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    if (myfile.is_open())
+    {
+        while (getline(myfile, line, ':') && getline(myfile, date, ':') && getline(myfile, bestTries))
+        {
+            if (currentLine == lineNumber)
+            {
+                if (!currentDate.empty() && date == currentDate) {
+                    std::cout << "Sorry you've already paly today!" << std::endl;
+                    return "";
```
Comment (bohdanpc) [inline r_d13/utils.cpp:29]: no close here?

File: r_d13/utils.cpp [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+#include <string.h>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include "utils.h"
+#include "wordle.h"
+
+const int subtractor = 'a' - 'A';
+
+std::string readFromFile(const std::string filePath, int lineNumber, std::string currentDate)
```
Comment (bohdanpc) [inline r_d13/utils.cpp:12]: read from file what?

File: r_d13/codeGameTest.cpp [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    getline(cin, secret_word);
+    string user_guess;
+    getline(cin, user_guess);
+
+    
+    toUpperCase(secret_word);
+    toUpperCase(user_guess);
+
+    char result[WORD_SIZE + 1] = "*****";
+    compareWords(&secret_word[0], &user_guess[0], result);
```
Comment (bohdanpc) [inline r_d13/codeGameTest.cpp:75]: not like this.
secret_word.c_str() will return correct pointer

Comment (bohdanpc) [inline r_d13/codeGameTest.cpp:75]: and why not using std::string everywhere then? 😄

File: r_d13/r_d13.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+int main()
+{
+	bool isExit = false;
+	do {
+		int round = 0;
+		bool isFound = false;
+		char* word = nullptr;
+
+		showStart();
+		TypeOfWord type = inputTypeOfWord();
```
Comment (bohdanpc) [inline r_d13/r_d13.cpp:20]: const

File: r_d13/utils.cpp [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    }
+    else
+    {
+        std::cout << "Unable to open file" << std::endl;
+    }
+
+    return "";
+}
+
+bool writeToFile(const std::string filePath, std::string word, std::string currentDate, int tries)
```
Comment (bohdanpc) [inline r_d13/utils.cpp:46]: write what?

Comment (bohdanpc) [reply r_d13/utils.cpp:46]: pass all std::strings by const ref

File: r_d13/utils.cpp [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+{
+    std::fstream file(filePath, std::ios::in | std::ios::out);
```
Comment (bohdanpc) [inline r_d13/utils.cpp:48]: in and out at the same time?

File: r_d13/utils.cpp [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    std::ifstream inputFile(filePath);
+    std::ofstream tempFile("temp.txt");
```
Comment (bohdanpc) [inline r_d13/utils.cpp:50]: no way you need it)

File: r_d13/r_d13.cpp [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+		do {
+			isFound = gameRound(word, round);
+		} while (round != MAX_ROUND && !isFound);
+
+		showFinish(isFound, round);
+
+		time_t now = time(0);
+		tm* ltm = localtime(&now);
+		std::string monthDay = std::to_string(ltm->tm_mon) + "/" + std::to_string(ltm->tm_mday);
+		writeToFile(filePath, word, monthDay, round);
```
Comment (bohdanpc) [inline r_d13/r_d13.cpp:38]: You only need it for WorldeOfTheDay mode, not unconditionally

**Pull request merged**